sdk/core/azure-core/test/ut/curl_session_test.cpp

```diff
@@ -6,7 +6,8 @@
 #include <azure/core/http/curl/curl.hpp>
 #include <azure/core/http/http.hpp>
 
-#include <curl/curl.h>
+#include "../../src/http/curl/curl_connection.hpp"
```

Comment:
Is hardcoding the relative path like this maintainable?

Do we want our tests to depend on private header files? This is out of the scope of the PR, but maybe we should re-write some of the test to avoid this dependency.

Reply:
yeah, I don't really like this too much. But this is the way we can still test the private stuff (which needs to be tested anyway).
The other approach is to create a private lib from the CMake tests and consume it. I think that would keep the relative path in one place only (instead of at each include). I will make this change as well with the "private" suffix

Reply:
> But this is the way we can still test the private stuff (which needs to be tested anyway).

I think we should evaluate how we test the SDK. To me, we should focus most of our attention to testing things directly via the public surface area only (at least by default). The implementation and branches reachable through public API calls are what matter most, and we should be able to get to 95%+ coverage just trough that. For edge cases, where it is quite difficult to reach a code path, we can evaluate and make exceptions. This way, it is easier to see unreachable code paths and if a component or branch is not necessary/used by anything other than tests, we can remove or simplify them.

Tightly  coupling private implementation detail and tests makes it harder to change the implementation. Ideally, the tests encode the public API and its behavior and enforce that those things don't change. They shouldn't dictate the specifics of the implementation.

> The other approach is to create a private lib from the CMake tests and consume it. I think that would keep the relative path in one place only (instead of at each include). I will make this change as well with the "private" suffix

Maybe hold off on that one for now, and let's discuss it separately.